### ISSUE TITLE
`@stylistic/selector-pseudo-class-parentheses-space-inside` reports a drifting position, and rewrites the comments it fixes past, in a selector carrying an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#163]
Found while fixing [#161](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/161) and [#128](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/128), and older than either: this rule reads a selector from a copy of it that `postcss-scss` spells differently, and writes its fix to the copy that is not printed.

### What minimal example or steps are needed to reproduce the bug?

```scss
a:not(
	// c
) {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-pseudo-class-parentheses-space-inside": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`a:not( /* c */ )`, or whatever the option asks for, written to the file — and a warning standing on the closing parenthesis while it is not, `3:1`.

### What actually happened?

Two things, and each is enough on its own.

**The fix never arrives.** `--fix` leaves the file exactly as it was and reports a clean pass, so the problem survives every run:

```
1:7  Expected single space after "("
3:2  Expected single space before ")"
```

Run `--fix` and the file is untouched; lint it again and both warnings are back. Under `always` that is [#128](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/128) all over again, in the one place its fix does not reach: a warning that outlives its own fix, for as long as the file holds an inline comment there.

**The position is not the position of the file.** The closing warning stands at `3:2`, one character past the end of that line, where nothing stands at all. The drift is two characters per comment, so it grows with every one of them and walks into the middle of the code it is about:

```scss
a:not(
	// one
	// two
	b) {}
```

```
4:6  Expected single space before ")"   /* line 4 holds `\tb) {}`, and column 6 is the closing brace */
```

### Do you have a proposal to fix the bug?

`postcss-scss` rewrites every inline comment of a selector into a block comment inside `raws.selector.raw`, and keeps the source spelling in `raws.selector.scss`, which is the copy its stringifier prints. `/* c*/` is two characters longer than `// c`, so every index counted in the raw stands two characters further along per comment than the file spells it, and a fix written to the raw is never printed.

`selectorCombinatorSpaceChecker` already answers both halves for `selector-combinator-space-before` and `selector-combinator-space-after`: `findInlineComments` pairs every inline comment of the parsed selector with the text the source gives it, `toSourceIndex` translates an index of the raw into an index of the file, and the fix is written to `raws.selector.scss` as well as to the raw, with the comments given their `//` spelling back. Both helpers are private to that file today. Lifting them into `lib/utils/` and reading and writing this rule through them is the whole of the fix, and it is the same machinery [#158](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/158) wants for `selector-descendant-combinator-no-non-space` — so the three want doing together, or at least the extraction does.

**Comment by firefoxic** [comment]
The branch that fixes #161 and #128 passes such a selector over rather than leaving it as it was, so what this issue asks for is unchanged but its symptom is not: nothing is reported in a selector carrying an inline comment under this syntax, instead of a warning standing where the file has no such character and a fix that never arrives.

The reason for taking that measure now, rather than leaving the reports standing until this is answered: giving the parsed selector back the whitespace the parser drops beside a comment — which is what #161 and #128 needed — makes that whitespace visible to the `never` options as well, and under this syntax their fix cannot reach the file. `a:not( /* c */ ) // t` would have drawn a second warning that no number of runs could clear. A selector this rule cannot write to is one it has nothing to say about, so it says nothing, as `selector-descendant-combinator-no-non-space` already does for #158.

The work this issue asks for is unchanged: read the positions through a translation from the raw to the file, and write the fix to both copies, with the inline comments given their `//` spelling back.
